docstr_coverage/cli.py

```diff
@@ -328,7 +346,21 @@ def execute(paths, **kwargs):
     show_progress = not kwargs["percentage_only"]
     results = analyze(all_paths, ignore_config=ignore_config, show_progress=show_progress)
 
-    LegacyPrinter(verbosity=kwargs["verbose"], ignore_config=ignore_config).print(results)
+    report_format = kwargs["format"]
+    if report_format == "markdown":
+        printer = MarkdownPrinter(results, verbosity=kwargs["verbose"], ignore_config=ignore_config)
+    elif report_format == "text":
+        printer = LegacyPrinter(results, verbosity=kwargs["verbose"], ignore_config=ignore_config)
+    else:
+        raise SystemError("Unknown report format: {0}".format(report_format))
+
+    output_type = kwargs["output"]
+    if output_type == "file":
+        printer.save_to_file()
```

Comment:
Haven't checked in detail yet, but is the legacy printer supporting file? And is the markdown printer supporting stdout? 

If not, we should raise suitable errors here. 

That said, I would see value of putting the Legacy output to a text file. (Markdown to stdout not so much, but it would not hurt having it, in case I missed that it's already supported :-D )

Reply:
For now Legacy and Markdown are supporting `stdout` and `file` output because of their base `Printer` class with corresponding abstract methods. As I said https://github.com/HunterMcGushion/docstr_coverage/pull/145#issuecomment-2366913498 this feature can be changed.